controllers/amphoracontroller_controller.go

```diff
@@ -463,40 +464,42 @@ func (r *OctaviaAmphoraControllerReconciler) reconcileNormal(ctx context.Context
 		return ctrlResult, nil
 	}
 
-	// verify if network attachment matches expectations
-	networkReady, networkAttachmentStatus, err := nad.VerifyNetworkStatusFromAnnotation(
-		ctx,
-		helper,
-		instance.Spec.NetworkAttachments,
-		serviceLabels,
-		instance.Status.ReadyCount,
-	)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+	if dset.GetDaemonSet().Generation <= dset.GetDaemonSet().Status.ObservedGeneration {
```

Comment:
I would use == and let the system reconcile again if < as that means something pushed the generation of the Spec to newer than what this controller wants